### PR TITLE
739 - Combine Library Metadata

### PR DESCRIPTION
**Pull request by avrohomgottlieb**
## Issue Number

https://github.com/AlexsLemonade/scpca-portal/issues/739

## Purpose/Implementation Notes

- Add `get_metadata` methods to `Project`, `Sample` and `Library` models which collect all metadata fields relevant to each model, whether in the model's general attributes or in the models `(additional_)metadata` attribute, and returns then in a dictionary
- Add `Library::get_combined_library_metadata` method which combines all of these dictionaries on a per library basis and returns a list of a single dictionary (or of multiple dictionaries if the library is multiplexed).

## Types of changes

What types of changes does your code introduce?

- New feature (non-breaking change which adds functionality)

## Functional tests

N/A

## Checklist

- [x] Lint and unit tests pass locally with my changes

## Screenshots

N/A

### CODE REVIEW

File: api/scpca_portal/models/sample.py [hunk 1]
```diff
@@ -120,6 +120,33 @@ def bulk_create_from_dicts(
...
+        sample_metadata.update(
+            {key: getattr(self, key) for key in dict(self) if key in included_sample_attributes}
+        )
+
+        sample_metadata.update(
+            {key: self.additional_metadata[key] for key in self.additional_metadata}
+        )
+
+        return sample_metadata
+
```
Comment (davidsmejia) [inline api/scpca_portal/models/sample.py:149]: When you write a ticket about updating how we handle sample metdata / moving away from `sample.additional_metadata` and towards `sample.metadata`. Include this method as a reason why.

Just as a reminder we want to move toward thinking about these attribtues as one of the following:
- metadata (raw inputs from s3)
- attributes (denormalized metadata keys)
- aggregate values (populated after we do bulk create of a related model)

Then we can think about this data in two contexts: persistence / querying and presentation.

This function should be excluded keys (if any) in the same way we do that for `Library.get_metadata`

**Review by davidsmejia — COMMENTED** [review]
Looks good to me, just the one change for comment and where we remove the `submitter` from the metadata_file module and remove it where we *currently* remove project metadata keys.

Otherwise I'm just requesting that you open a ticket to remind us to handle sample and project (additional) metadata uniformly as outlined in my comment.

After those items are complete this looks good to merge!

**Comment by avrohomgottlieb** [comment]
As referenced in Issue https://github.com/AlexsLemonade/scpca-portal/issues/763, I've added support for `filtered_spots`, `unfiltered_spots`, and `tissue_spots` to be included in the output metadata tsv file. This will affect only `Spatial` libraries. As a note, I did not adjust the old (current) implementation, which is on its way out, but removed these keys from the `excluded_metadata_attributes` list in the `Library::get_metadata` method which will include them in the output metadata tsv file in the new implementation.

File: api/scpca_portal/models/sample.py (hunk 1 above)
Comment (avrohomgottlieb) [reply api/scpca_portal/models/sample.py:149]: Here is the described issue: https://github.com/AlexsLemonade/scpca-portal/issues/774

File: api/scpca_portal/models/project.py [hunk 2]
```diff
@@ -786,6 +793,10 @@ def load_samples_metadata(self, sample_id: str = None) -> List[Dict]:
 
             self.add_project_metadata(sample_metadata)
 
+            # This is a temporary fix until we get rid of `load_samples_metadata` imminently.
```
Comment (davidsmejia) [inline api/scpca_portal/models/project.py:796]:
```suggestion
            # TODO: This is a temporary fix until we get rid of `load_samples_metadata` imminently.
```

**Review by davidsmejia — APPROVED** [review]
LGTM just one suggestion selfishly so I will see this highlighted.

**Pull request merged**